app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java

```diff
@@ -70,36 +70,51 @@
 
   private Unbinder unbinder;
 
+  private String zimFileName;
   private String zimFileTitle;
   private String articleTitle;
-  private String zimNoteDirectoryName;  // Corresponds to "ZimFileName" of "{External Storage}/Kiwix/Notes/ZimFileName/ArticleUrl.txt"
-  private String articleNotefileName;   // Corresponds to "ArticleUrl" of "{External Storage}/Kiwix/Notes/ZimFileName/ArticleUrl.txt"
+  // Corresponds to "ZimFileName" of "{External Storage}/Kiwix/Notes/ZimFileName/ArticleUrl.txt"
+  private String zimNoteDirectoryName;
+  // Corresponds to "ArticleUrl" of "{External Storage}/Kiwix/Notes/ZimFileName/ArticleUrl.txt"
+  private String articleNotefileName;
   private boolean noteFileExists = false;
-  private boolean noteEdited = false;   // Keeps track of state of the note (whether edited since last save)
+  boolean noteEdited = false;   // Keeps track of state of the note (whether edited since last save)
 
-  private String ZIM_NOTES_DIRECTORY;   // Stores path to directory for the currently open zim's notes
+  private String ZIM_NOTES_DIRECTORY; // Stores path to directory for the currently open zim's notes
 
-  public AddNoteDialog(SharedPreferenceUtil sharedPreferenceUtil) {
+  public AddNoteDialog(@NonNull SharedPreferenceUtil sharedPreferenceUtil) {
     this.sharedPreferenceUtil = sharedPreferenceUtil;
   }
 
   @Override
-  public void onCreate(Bundle savedInstanceState) {
+  public void onCreate(@Nullable Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
 
-    setStyle(DialogFragment.STYLE_NORMAL, sharedPreferenceUtil.nightMode() ? R.style.AddNoteDialogStyle_Night : R.style.AddNoteDialogStyle);
+    setStyle(DialogFragment.STYLE_NORMAL,
+        sharedPreferenceUtil.nightMode() ? R.style.AddNoteDialogStyle_Night
+            : R.style.AddNoteDialogStyle);
+
+    // Returns name of the form ".../Kiwix/granbluefantasy_en_all_all_nopic_2018-10.zim"
+    zimFileName = ZimContentProvider.getZimFile();
 
-    zimFileTitle = ZimContentProvider.getZimFileTitle();
-    articleTitle = ((MainActivity)getActivity()).getCurrentWebView().getTitle();
+    if (zimFileName != null) { // No zim file currently opened
+      zimFileTitle = ZimContentProvider.getZimFileTitle();
+      articleTitle = ((MainActivity) getActivity()).getCurrentWebView().getTitle();
 
-    zimNoteDirectoryName = getZimNoteDirectoryName();
-    articleNotefileName = getArticleNotefileName();
+      zimNoteDirectoryName = getZimNoteDirectoryName();
+      articleNotefileName = getArticleNotefileName();
 
-    ZIM_NOTES_DIRECTORY = NOTES_DIRECTORY + zimNoteDirectoryName + "/";
+      ZIM_NOTES_DIRECTORY = NOTES_DIRECTORY + zimNoteDirectoryName + "/";
+    } else {
+      showToast(R.string.error_filenotfound, Toast.LENGTH_LONG);
+      closeKeyboard();
+      getFragmentManager().beginTransaction().remove(AddNoteDialog.this).commit();
```

Comment:
the method you want is `dismiss` no need for a fragment transaction I think

Reply:
This app crash occurs only if you tap 'Clear All Notes' for the very first time, that too because for some reason 'Add Note Dialog' is called (The code I wrote doesn't do that)

Probably due to this edge case nature of the crash, dismiss() didn't close the dialog - it continued with setting up the dialog

Worked with transaction, so I used it

Reply:
transactions are asynchronous I guess - unless you use `commitNow` - so it gives it enough time to be added? Odd